src/praisonai/praisonai/ui_chat/default_app.py

```
@@ -39,11 +49,53 @@ async def on_welcome():
+                llm=os.getenv("MODEL_NAME", "gpt-4o-mini"),
+                tools=tools if tools else None,
+            )
+        except ImportError:
+            # Fallback to OpenAI if PraisonAI agents not available
+            _agent = None
+    return _agent
+
 @aiui.reply
-async def on_message(message: str):
-    """Stream a response from OpenAI."""
+async def on_message(message: str, settings: dict = None):
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)